services/sqs/src/main/java/software/amazon/awssdk/services/sqs/internal/SqsExecutionInterceptor.java

```diff
@@ -42,16 +41,11 @@
     @Override
     public SdkHttpFullRequest modifyHttpRequest(Context.ModifyHttpRequest context, ExecutionAttributes executionAttributes) {
         SdkHttpFullRequest request = context.httpRequest();
-        URI endpoint = request.getEndpoint();
 
         // If the request is using a non-standard endpoint, then
         // alter it to use the corresponding, standard endpoint
-        if (NONSTANDARD_ENDPOINT_MAP.containsKey(endpoint.getHost())) {
-            String newHost = NONSTANDARD_ENDPOINT_MAP.get(endpoint.getHost());
-            String newEndpoint = endpoint.toString().replaceFirst(endpoint.getHost(), newHost);
-            return request.toBuilder()
-                          .endpoint(URI.create(newEndpoint))
-                          .build();
+        if (NONSTANDARD_ENDPOINT_MAP.containsKey(request.host())) {
```

Comment:
I'd be in favor of removing this interceptor altogether.

Reply:
Sure, if it's safe. Not sure what it's for.